tests/CreateHabit.test.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+import renderer from 'react-test-renderer';
+import configureStore from 'redux-mock-store';
+
+const mockStore = configureStore([]);
+
+import CreateHabit from '../src/components/CreateHabit';
+
+describe('<CreateHabit />', () => {
+
+    let store;
+
+    beforeEach(() => {
+        store = mockStore({ count: { habits: {} } });
+    });
+
+
+    it('has 1 child', () => {
+        const tree = renderer.create(<Provider store={store}><CreateHabit /></Provider>).toJSON();
+        expect(tree.children.length).toBe(1);
+    });
+
+    it('renders correctly', () => {
+        const tree = renderer.create(<Provider store={store}><CreateHabit /></Provider>).toJSON();
+        expect(tree).toMatchSnapshot();
+    })
```

Comment:
With `react-test-renderer`, can you get the props and call actual functions? Would we be able to test that the store is modified if we complete the modal and submit?

Reply:
@jdbann - Have been looking at this and implemented some more unit test using react-testing-library/native - allows us to test buttons etc. There are a few drawbacks - having some issues correctly identifying react native paper components and the unit tests aren't as streamlined as I would like. Might be something to look at in our next pairing session

Reply:
Oh wow, this looks great! How does it feel as a test to you? Over-the-top or useful extra assurance? Do you feel inclined to add more of these as we go?

Reply:
I think this will be very useful for testing as we add more components on to the page. Some of the functionality feels like it is being tested twice - both here and in the reducers tests where we look at store changes. Might be worth streamlining these ones to just check a function is called rather that the outcome and the outcome is tested by the reducer. I'll keep working on it as we go.